[READY] Fix unicode warning when jumping on Python 2

micbou opened the pull request:
On Python 2, jumping in a file whose path contains non-ASCII characters raises the following warnings:
```
Error detected while processing function <SNR>26_CompleterCommand:
line   18:
python\ycm\vimsupport.py:400: UnicodeWarning: Unicode unequal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if filename != GetCurrentBufferFilepath():
python\ycm\vimsupport.py:375: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if win.buffer.name == filepath:
```
This happens because, in the `JumpToLocation` function, we are comparing a unicode object (`filename` and `filepath`) with a Vim buffer name (`vim.current.buffer.name` returned by `GetCurrentBufferFilepath` and `win.buffer.name` where `win` is a Vim window) which is a byte object on Python 2.

For now, this PR adds tests covering the `JumpToLocation` function with unicode paths. They will raise the warnings on Python 2 and will fail because warnings are now treated as errors in tests. I'll update the PR with the fix once the builds are done.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/valloric/youcompleteme/2702)
<!-- Reviewable:end -->

micbou commented:
I've pushed the fix. Tests are now passing.

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/valloric/youcompleteme/2702#-:-KoFBO-R7oOe042_DX8j:b-xfeuwf)*
<!-- Sent from Reviewable.io -->

bstaletic commented:
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[python/ycm/tests/test_utils.py, line 237 at r1](https://reviewable.io:443/reviews/valloric/youcompleteme/2702#-KoGNbP_ob4KEC_wgDGl:-KoGNbP_ob4KEC_wgDGm:b-lajwci) ([raw file](https://github.com/valloric/youcompleteme/blob/e5734575636868e31889b1eb1fd1fd8455a3d7f4/python/ycm/tests/test_utils.py#L237)):*
> ```Python
>     self.modified = modified
>     self.hidden = hidden
>     self.window = window if not hidden else None
> ```

Hidden doesn't mean a buffer has no window. If `hidden` is set the buffer can be hidden and without a window. In my `vimrc` I have a line saying `set hidden` which makes `:echo &hidden` return `1` for all windows.

So are we sure we should do this? Is this part of the logic even needed?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/valloric/youcompleteme/2702)*
<!-- Sent from Reviewable.io -->

micbou commented:
Reviewed 2 of 3 files at r2.
Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[python/ycm/tests/test_utils.py, line 237 at r1](https://reviewable.io:443/reviews/valloric/youcompleteme/2702#-KoGNbP_ob4KEC_wgDGl:-KoHIIka7batj-qZ8mSq:b-ji48j9) ([raw file](https://github.com/valloric/youcompleteme/blob/e5734575636868e31889b1eb1fd1fd8455a3d7f4/python/ycm/tests/test_utils.py#L237)):*
<details><summary><i>Previously, bstaletic wrote…</i></summary><blockquote>

Hidden doesn't mean a buffer has no window. If `hidden` is set the buffer can be hidden and without a window. In my `vimrc` I have a line saying `set hidden` which makes `:echo &hidden` return `1` for all windows.

So are we sure we should do this? Is this part of the logic even needed?
</blockquote></details>

You are right. [`hidden`](http://vimdoc.sourceforge.net/htmldoc/options.html#'hidden') only controls if a buffer can be hidden or not. In addition, this is a global option. The local one is named [`bufhidden`](http://vimdoc.sourceforge.net/htmldoc/options.html#'bufhidden'). I've updated the `HiddenEnabled` function accordingly and added a few tests. I've also changed the `modified` buffer option to be false by default (no changes in the buffer).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/valloric/youcompleteme/2702)*
<!-- Sent from Reviewable.io -->

bstaletic commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 3 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/valloric/youcompleteme/2702#-:-KoIbQGQU0LJZJKVg4GA:bnfp4nl)*
<!-- Sent from Reviewable.io -->

vheon commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> @zzbot r=bstaletic 

---

Reviewed 1 of 4 files at r1, 2 of 3 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/valloric/youcompleteme/2702#-:-KoJRFgaOaGqoDog0k1_:b-43p6gy)*
<!-- Sent from Reviewable.io -->
